src/components/editor/Bindings/index.tsx

```diff
@@ -60,35 +73,59 @@ function BindingsMultiEditor({ draftSpecs = [], readOnly = false }: Props) {
     const resourceConfigUpdated = useMemo(() => {
         let queriedResourceConfig: ResourceConfigDictionary = {};
 
+        const collectionNameProp =
+            entityType === ENTITY.MATERIALIZATION ? 'source' : 'target';
+
         draftSpecs[0]?.spec.bindings.forEach((binding: any) => {
             queriedResourceConfig = {
                 ...queriedResourceConfig,
-                [binding.source]: {
+                [binding[collectionNameProp]]: {
                     data: binding.resource,
                     errors: [],
                 },
             };
         });
 
-        // TODO (optimization): Evaluate the performance of a hash comparator function.
         return draftSpecs.length > 0
             ? !isEqual(resourceConfig, queriedResourceConfig)
             : false;
-    }, [draftSpecs, resourceConfig]);
+    }, [draftSpecs, entityType, resourceConfig]);
 
     useEffect(() => {
-        if (workflow === 'materialization_edit') {
-            setServerUpdateRequired(resourceConfigUpdated);
-        }
-    }, [setServerUpdateRequired, resourceConfigUpdated, workflow]);
+        setServerUpdateRequired(resourceConfigUpdated);
+    }, [setServerUpdateRequired, resourceConfigUpdated]);
+
+    const { liveSpecs } = useLiveSpecs('collection');
+
+    const fetchingSpecs =
+        entityType === ENTITY.MATERIALIZATION
+            ? liveSpecs.length === 0
+            : draftSpecs.length === 0;
```

Comment:
PENDING: This condition is inaccurate for captures. Whether the draft specs _and_ live specs data is undergoing validation is the determinizing factor. Additionally, draft spec and live spec data should be passed to the `CollectionPicker` component via props to prevent duplicate API calls.